packages/integrations/vercel/src/index.ts

```diff
@@ -33,21 +33,39 @@ export default function vercel(): AstroIntegration {
 			'astro:build:start': async ({ buildConfig }) => {
 				buildConfig.serverEntry = `${ENTRYFILE}.mjs`;
 				buildConfig.client = new URL('./static/', _config.outDir);
-				buildConfig.server = new URL('./server/pages/', _config.outDir);
+				buildConfig.server = new URL('./server/out/', _config.outDir);
 			},
 			'astro:build:done': async ({ dir, routes }) => {
-				const pagesDir = new URL('./server/pages/', dir);
+				/*
+					Why do we have two folder? Why don't we just generate all inside server/pages?
+					When the app builds, it throws some metadata inside a `chunks/` folder.
+
+					./server/
+						pages/
+							__astro_entry.mjs
+							chunks/
+								(lots of js files)
+					
+					Those chunks will count as serverless functions (which cost money), so we
+					need to bundle as much as possible in one file. Hence, the following code
+				*/
+
+				const serverOutDir = new URL('./server/out/', dir);
```

Comment:
This is a small nit, but `tmp` might make more sense here.

```suggestion
				const serverTmpDir = new URL('./server/tmp/', dir);
```